pkg/updatechecker/updatechecker.go

```diff
@@ -472,21 +472,23 @@ func autoDeploy(opts CheckForUpdatesOpts, clusterID string, autoDeploy apptypes.
 	}
 
 	currentVersion := appVersions.CurrentVersion
-	if currentVersion == nil || currentVersion.Semver == nil {
+	if currentVersion == nil {
 		return nil
 	}
 
 	var versionToDeploy *downstreamtypes.DownstreamVersion
 
-	// In the case semver is not required, we only need to check if the newest app version
-	// is greater than the current version.
 	if autoDeploy == apptypes.AutoDeploySequence {
-		if currentVersion.Sequence < appVersions.AllVersions[0].Sequence {
+		// semver is not required/enabled, we only need to check if the newest app version is newer than the current version.
+		// use cursor instead of sequence in order to only deploy newer upstream versions, and not versions created by config changes, license changes, etc...
+		currentCursor := currentVersion.Cursor
+		latestCursor := appVersions.AllVersions[0].Cursor
+		if currentCursor != nil && latestCursor != nil && (*currentCursor).Before(*latestCursor) {
```

Comment:
Should this be determined by that weird sorting algorithm we have?

Reply:
that algorithm will sort using sequences and not cursors, which is what causes the issue of automatically deploying config changes, license changes, etc...

this code block is only relevant for when semantic versioning is disabled any ways

Reply:
also, `appVersions.AllVersions` is already sorted using that algorithm

Reply:
also, `appVersions.AllVersions[0]` is the top version after sorting, so it for sure is higher than or equal to the current version in terms of order. then the only question becomes whether or not it has a higher cursor if it's a variation of the same upstream version.